wolfcrypt/src/signature.c

```diff
@@ -175,7 +175,15 @@ int wc_SignatureVerify(
                 int is_valid_sig = 0;
 
                 /* Perform verification of signature using provided ECC key */
-                ret = wc_ecc_verify_hash(sig, sig_len, hash_data, hash_len, &is_valid_sig, (ecc_key*)key);
+                do {
+                #ifdef WOLFSSL_ASYNC_CRYPT
+                    ret = wc_AsyncWait(ret, &((ecc_key*)key)->asyncDev,
```

Comment:
Its already on the list of future enhancements. When I wrote the signature wrapper it was designed without a context, so current API's cannot support support returning WC_PENDING_E. I agree adding new async API's for the signature wrapper is a good idea.